src/IterableObject.php

```diff
@@ -27,15 +27,27 @@ final class IterableObject implements IteratorAggregate
     /**
      * @param iterable<mixed>|null $iterable
      */
-    public function __construct(?iterable $iterable = null, ?callable $filter = null, ?callable $map = null)
+    private function __construct(?iterable $iterable = null, ?callable $filter = null, ?callable $map = null)
     {
         $this->iterable = $iterable ?? new EmptyIterator();
         $this->filterFn = $filter;
         $this->mapFn = $map;
     }
 
-    public function filter(callable $filter): self
+    /**
+     * @param iterable<mixed>|null $iterable
+     */
+    public static function new(?iterable $iterable = null): self
     {
+        return new self($iterable);
+    }
+
+    public function filter(?callable $filter = null): self
+    {
+        $filter = $filter ?? function ($value): bool {
```

Comment:
When I did the benchmarking last time on 7.4, using `intval` or `boolval` in this case was faster than anon. fcn.

Reply:
Good idea 🙂 but I just tested it, and it throws warning when used within `CallbackFilterIterator` because it expects 1 parameter, and the `CallbackFilterIterator` injects 3.

Reply:
That reminds me that we currently don't support the `$mode` argument of the [array_filter()](https://www.php.net/manual/fr/function.array-filter.php) function, I'll try to address that in a separate PR.